src/performTemplateReplacements.js

```diff
@@ -0,0 +1,179 @@
+// (C) 2021 GoodData Corporation
+
+import { getSchema, DEFAULT_SCHEMA, getHostnameWithSchema } from "./stringUtils";
+import replaceInFiles from "./replaceInFiles";
+
+export const performTemplateReplacements = async ({
+    targetDir,
+    sanitizedAppName,
+    hostname,
+    backend,
+    flavor,
+}) => {
+    const hostnameSchema = getSchema(hostname) || DEFAULT_SCHEMA;
+
+    const isTigerBackend = backend === "tiger";
+    const isJsFlavor = flavor === "js";
+
+    // this object has structure corresponding to the file structure relative to targetDir
+    // having it like this makes sure that all the replacements relevant to each file are in one place, thus preventing race conditions
+    const replacementDefinitions = {
+        "package.json": [
+            { regex: /@gooddata\/gdc-app-name/, value: sanitizedAppName, apply: true },
+            {
+                regex: /@gooddata\/sdk-backend-bear/g,
+                value: "@gooddata/sdk-backend-tiger",
+                apply: isTigerBackend,
+            },
+            {
+                regex: /"refresh-ldm": "node .\/scripts\/refresh-ldm.js"/g,
+                value: '"refresh-ldm": "node ./scripts/refresh-ldm.js --backend tiger"',
+                apply: isTigerBackend,
+            },
+            {
+                regex: /"start": "cross-env HTTPS=true react-scripts start",/g,
+                value: '"start": "react-scripts start",',
+                apply: hostnameSchema !== "https",
+            },
+            {
+                regex: /\n.*@types\/.*/g, // match also previous new line to not have blank spaces between dependencies
```

Comment:
Nitpick: this would probably bomb if a `@types` package was last in a group, then the comma at the end of the previous entry would invalidate the JSON. However, we should keep them sorted so the risk of this being a problem is super low, feel free to skip this.

Reply:
Yep, I originally removed TypeScript package as well and it was the last one. I was deleting the comma as well to fix the exact issue you described. However, I returned the TS package back because it was required by one dependency anyway. Each time you add dependency, yarn (maybe even npm) will put it to the correct alphabetical order so this should not happen unless someone removes all the subsequent dependencies. He will probably see the issue right when he tries to build the app. I will skip this.

Reply:
Fair enough :)